File: python/interviewbit/arrays/MinStepsInInfinityGrid.py
class Solution:
    # @param A : list of integers
    # @param B : list of integers
    # @return an integer
    def coverPoints(self, A, B):
        if len(A) == 0:
            return 0;
        steps = 0;
        current_x = A[0];
        current_y = B[0];
        for i in range(1, len(A)):
            steps += max(abs(A[i] - current_x), abs(B[i] - current_y));
            current_x = A[i];
            current_y = B[i];
        return steps;

File: js/interviewbit/arrays/MinStepsInInfinityGrid.js
module.exports = {
        //param A : array of integers
        //param B : array of integers
        //return an integer
        coverPoints : function(A, B){
        if (A.length === 0) return 0;
        var steps = 0;
        var currentX = A[0];
        var currentY = B[0];
        for (var i = 1; i < A.length; ++i) {
            steps += Math.max(Math.abs(A[i] - currentX), Math.abs(B[i] - currentY));
            currentX = A[i];
            currentY = B[i];
        }
        return steps;
    }
};